src/ORM/DataObject.php

```diff
@@ -1643,6 +1643,11 @@ public function writeComponents($recursive = false, $skip = [])
      */
     private function skipWriteComponents($recursive, DataObject $target, array &$skip)
     {
+        // skip writing component if it doesn't exist
+        if (!$target->exists()) {
```

Comment:
Just want to point out that `exists` doesn't actually hit the DB. It just check if the object has an ID and return `false` if it doesn't.

I don't think that's problem. That sounds like a fair assumption in this context.